### ISSUE TITLE
Question: SVG rendering

### ISSUE THREAD

**Issue by itsallmememe** [rainbow-me/rainbow#4482]
I'd like to know how you implement your NFT SVG rendering? 

I'm making a native iOS application  that is downloading NFT data services such as Simple Hash and NFT Port. They supply SVG assets in raw xml format and so I render these on the device. However most of the solutions are not that performant or are prone to rendering errors, or worse, crashes.

So, do you render SVGs to a png/jpeg on your API before displaying them in the application or do you use a library / a React Native library to render them on device? And if the latter, which one?!

Many thanks

**Comment by skylarbarrera** [comment]
hey @itsallmememe,

all this info should be pretty apparent when looking through the codebase. but here's a quick summary 👇 

previews are generated into jpegs

larger assets are displayed in a native web view

**Comment by itsallmememe** [comment]
@skylarbarrera - thanks for your quick response. We have looked at your dependencies and indeed sniffed you app to see what API's you're hitting. I'm no ReactNative expert tho, hence my questioning :)

We assumed you were using `react-svgr` for SVG rendering, and so I was looking for confirmation on that. 

> previews are generated into jpegs

Are you using the raw SVG xml from SimpleHash / your own API and running this through `react-svgr` to generate an image?

> larger assets are displayed in a native web view

Can I ask why you are using two different techniques? I've found that the native UIWebView is not ideal performance wise, but generally displays and handles malformed SVG's very well. Was this the consideration to this dual rendering solution?

Many thanks!
